### PR TITLE
JBPM-4895: Console: null column pre-selected when creating new Tasks …

### PR DESCRIPTION
**Pull request by nmirasch**
…filter

### CODE REVIEW

File: jbpm-console-ng-generic/jbpm-console-ng-data-filters-client/src/main/java/org/jbpm/console/ng/df/client/popup/filter/DataSetFilterEditor.java [hunk 1]
```diff
@@ -88,6 +87,8 @@ public void init(DataSetMetadata metadata, DataSetFilter filter, Listener listen
         initNewFilterListBox();
         filterListPanel.clear();
 
+        /*
```
Comment (Salaboy) [inline jbpm-console-ng-generic/jbpm-console-ng-data-filters-client/src/main/java/org/jbpm/console/ng/df/client/popup/filter/DataSetFilterEditor.java]: other reviewers will probably complain about commented code..

**Comment by Salaboy** [comment]
It sounds like this test should go in dash builder? or the DataSetFilterEditor is bpm specific?

File: jbpm-console-ng-generic/jbpm-console-ng-data-filters-client/src/test/java/org/jbpm/console/ng/df/client/popup/filter/DataSetFilterEditorTest.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    @Test
+    public void initEmptyListFilterTest() {
+        //After initialization want to ckeck if existing DataSet Filter Editor filter List panel is empty
+        DataSetFilter filter = new DataSetFilter();
+        dataSetFilterEditor.init(metadata, filter, null);
+        assert (!dataSetFilterEditor.filterListPanel.iterator().hasNext());
+
+    }
+    @Test
+    public void initEmptyFilterListWithoutShowilterWithExistingTest() {
```
Comment (triceo) [inline jbpm-console-ng-generic/jbpm-console-ng-data-filters-client/src/test/java/org/jbpm/console/ng/df/client/popup/filter/DataSetFilterEditorTest.java]: ```Showilter``` looks like a typo to me.

**Comment by nmirasch** [comment]
This code is related with the DatasetFilterEditor specific for the console. Just to allow create new filters, not to modify existing ones.

File: jbpm-console-ng-generic/jbpm-console-ng-data-filters-client/src/test/java/org/jbpm/console/ng/df/client/popup/filter/DataSetFilterEditorTest.java (hunk 2 above)
Comment (nmirasch) [inline jbpm-console-ng-generic/jbpm-console-ng-data-filters-client/src/test/java/org/jbpm/console/ng/df/client/popup/filter/DataSetFilterEditorTest.java]: you are right, I'm going to fix that. Thanks!

File: jbpm-console-ng-generic/jbpm-console-ng-data-filters-client/src/test/java/org/jbpm/console/ng/df/client/popup/filter/DataSetFilterEditorTest.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2015 JBoss by Red Hat.
```
Comment (cristianonicolai) [inline jbpm-console-ng-generic/jbpm-console-ng-data-filters-client/src/test/java/org/jbpm/console/ng/df/client/popup/filter/DataSetFilterEditorTest.java]: Please update to use the latest copyright version.

Copyright 2016 Red Hat, Inc. and/or its affiliates.

File: jbpm-console-ng-generic/jbpm-console-ng-data-filters-client/src/test/java/org/jbpm/console/ng/df/client/popup/filter/DataSetFilterEditorTest.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    public void setup() {
+        dataSetFilterEditor = new DataSetFilterEditor();
+    }
+
+    @Test
+    public void initEmptyListFilterTest() {
+        //After initialization want to ckeck if existing DataSet Filter Editor filter List panel is empty
+        DataSetFilter filter = new DataSetFilter();
+        dataSetFilterEditor.init(metadata, filter, null);
+        assert (!dataSetFilterEditor.filterListPanel.iterator().hasNext());
```
Comment (cristianonicolai) [inline jbpm-console-ng-generic/jbpm-console-ng-data-filters-client/src/test/java/org/jbpm/console/ng/df/client/popup/filter/DataSetFilterEditorTest.java]: I would recommend using assertFalse instead to make it a bit easier to read.

**Comment by cristianonicolai** [comment]
Looks good to me, just some minor update in the test class.

File: jbpm-console-ng-generic/jbpm-console-ng-data-filters-client/src/test/java/org/jbpm/console/ng/df/client/popup/filter/DataSetFilterEditorTest.java (hunk 4 above)
Comment (jhrcek) [inline jbpm-console-ng-generic/jbpm-console-ng-data-filters-client/src/test/java/org/jbpm/console/ng/df/client/popup/filter/DataSetFilterEditorTest.java]: Moreover assert is plain java keyword (not related to JUnit) and it only works when jvm is started with "-enableassertions" or "-ea" keywords, which will normally not be the case when running tests via maven/surefire.

**Comment by cristianonicolai** [comment]
+1

**Comment by csadilek** [comment]
master: 48c83c406ba24c32e2c5971ba31c6399b4867ecf

**Pull request closed without merging**